### PR TITLE
Add types for more RxJS 5 methods

### PR DESCRIPTION
**Pull request by matthewwithanm**
Types for the following methods:

- `audit()`
- `concatAll()`
- `pairwise()`
- `forkJoin()`

### CODE REVIEW

File: definitions/npm/rxjs_v5.0.x/flow_>=v0.29.0/rxjs_v5.0.x.js [hunk 1]
```diff
@@ -77,6 +79,8 @@ declare module 'rxjs' {
 
     concat(...sources: Observable<T>[]): Observable<T>;
 
+    concatAll(): T; // assumption: T is Observable
```
Comment (jeffmo) [inline definitions/npm/rxjs_v5.0.x/flow_>=v0.29.0/rxjs_v5.0.x.js]: Maybe do like `never` and use a local type bound to `Observable`?

```
concatAll<U>(): Observable<U>;
```

Comment (matthewwithanm) [inline definitions/npm/rxjs_v5.0.x/flow_>=v0.29.0/rxjs_v5.0.x.js]: 👍 Updated `mergeAll()` too.

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-v0.32.x/rxjs_v5.0.x.js [hunk 2]
```diff
@@ -23,6 +23,12 @@ interface rxjs$ISubscription {
 
 type TeardownLogic = rxjs$ISubscription | () => void;
 
+type EventListenerOptions = {
```
Comment (jeffmo) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-v0.32.x/rxjs_v5.0.x.js]: Really need to namespace these global types.
Not your doing though, so if you don't want to do it in this PR that's fine and I can do it in a follow-up.

Comment (matthewwithanm) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-v0.32.x/rxjs_v5.0.x.js]: That is my doing! One sec…

Comment (matthewwithanm) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-v0.32.x/rxjs_v5.0.x.js]: Also fixed TearDownLogic

**Pull request merged**